Gems/Atom/Utils/Code/Source/PngFile.cpp

```diff
@@ -0,0 +1,324 @@
+        }
+
+        PngFile PngFile::Load(const char* path, LoadSettings loadSettings)
+        {
+            if (!loadSettings.m_errorHandler)
+            {
+                loadSettings.m_errorHandler = [path](const char* message) { DefaultErrorHandler(AZStd::string::format("Could not load file '%s'. %s", path, message).c_str()); };
+            }
+
+            // For documentation of this code, see http://www.libpng.org/pub/png/libpng-1.4.0-manual.pdf chapter 3
+
+            FILE* fp = NULL;
+            if (fopen_s(&fp, path, "rb") || !fp)
+            {
+                loadSettings.m_errorHandler("Cannot open file.");
+                return {};
+            }
+
+            png_byte header[HeaderSize] = {};
+
+            if (fread(header, 1, HeaderSize, fp) != HeaderSize)
+            {
+                fclose(fp);
+                loadSettings.m_errorHandler("Invalid png header.");
+                return {};
+            }
+
+            bool isPng = !png_sig_cmp(header, 0, HeaderSize);
+            if (!isPng)
+            {
+                fclose(fp);
+                loadSettings.m_errorHandler("Invalid png header.");
+                return {};
+            }
+
+            png_voidp user_error_ptr = &loadSettings.m_errorHandler;
+            png_error_ptr user_error_fn = PngImage_user_error_fn;
+            png_error_ptr user_warning_fn = PngImage_user_warning_fn;
+
+            png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, user_error_ptr, user_error_fn, user_warning_fn);
+            if (!png_ptr)
+            {
+                fclose(fp);
+                loadSettings.m_errorHandler("png_create_read_struct failed.");
+                return {};
+            }
+
+            png_infop info_ptr = png_create_info_struct(png_ptr);
+            if (!info_ptr)
+            {
+                png_destroy_read_struct(&png_ptr, (png_infopp)NULL, (png_infopp)NULL);
+                fclose(fp);
+                loadSettings.m_errorHandler("png_create_info_struct failed.");
+                return {};
+            }
+
+            png_infop end_info = png_create_info_struct(png_ptr);
+            if (!end_info)
+            {
+                png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
+                fclose(fp);
+                loadSettings.m_errorHandler("png_create_info_struct failed.");
+                return {};
+            }
+
+#pragma warning(push)
+#pragma warning(disable: 4611) // Disables "interaction between '_setjmp' and C++ object destruction is non-portable". See https://docs.microsoft.com/en-us/cpp/preprocessor/warning?view=msvc-160
+            if (setjmp(png_jmpbuf(png_ptr)))
+            {
+                png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
+                fclose(fp);
+                // We don't report an error message here because the user_error_fn should have done that already.
+                return {};
+            }
+#pragma warning(pop)
```

Comment:
The `AZ_PUSH_DISABLE_WARNING` amd `AZ_POP_DISABLE_WARNING` macros should be used instead.

Reply:
This is now fixed. Change was required for Mac & iOS.

Reply:
Thanks!